modules/hydrodyn/src/HydroDyn_Input.f90

```diff
@@ -1191,42 +1165,24 @@ SUBROUTINE HydroDynInput_ProcessInitData( InitInp, Interval, InputFileData, ErrS
         
    
    ! WaveStMod - Model switch for stretching incident wave kinematics to instantaneous free surface.
-   IF ( InitInp%WaveMod /= 0 .AND. InputFileData%Morison%NMembers > 0 ) THEN
-      IF ( InitInp%WaveMod /= 6 ) THEN 
-         IF ( ( InitInp%WaveStMod /= 0 ) .AND. ( InitInp%WaveStMod /= 1 ) .AND. &
-              ( InitInp%WaveStMod /= 2 ) .AND. ( InitInp%WaveStMod /= 3 ) ) THEN
+   IF ( InitInp%WaveField%WaveMod /= WaveMod_None .AND. InputFileData%Morison%NMembers > 0 ) THEN
+      IF ( InitInp%WaveField%WaveMod /= WaveMod_ExtFull ) THEN 
+         IF ( ( InitInp%WaveField%WaveStMod /= 0 ) .AND. ( InitInp%WaveField%WaveStMod /= 1 ) .AND. &
+              ( InitInp%WaveField%WaveStMod /= 2 ) .AND. ( InitInp%WaveField%WaveStMod /= 3 ) ) THEN
             ErrMsg  = ' WaveStMod must be 0, 1, 2, or 3.'
             ErrStat = ErrID_Fatal
             RETURN
          END IF
       ELSE
-         IF ( ( InitInp%WaveStMod /= 0 ) .AND. ( InitInp%WaveStMod /= 1 ) .AND. &
-              ( InitInp%WaveStMod /= 3 ) ) THEN
+         IF ( ( InitInp%WaveField%WaveStMod /= 0 ) .AND. ( InitInp%WaveField%WaveStMod /= 1 ) .AND. &
```

Comment:
This check is different than the one in SeaSt_Input.f90/SeaStateInput_ProcessInitData(). In SeaState, it says WaveStMod must be 0 when WaveMod is WaveMod_ExtFull (6). Here in HydroDyn, it says it can be 0, 1, or 3. @luwang00 , do you know what the check should actually be?

Reply:
@bjonkman I believe the check performed in HydroDyn here is correct. Only extrapolation stretching (WaveStMod=2) cannot be used with full-field data. All other wave-stretching methods should work.

Reply:
Addressed in 67c77c: This check was moved to SeaState, to replace the conflicting check there (and removed from HydroDyn)